homework/index.js

```diff
@@ -29,19 +29,144 @@
     });
     return elem;
   }
+  /* cSpell:enable */
+  // Creating the repository-details part
+  function createHyfRepoDetails(leftDiv, hyfRepo) {
+    const table = createAndAppend('table', leftDiv, { class: 'table' });
+    const tBody = createAndAppend('tbody', table);
+    const details = ['Repository', 'Description', 'Forks', 'Updated'];
+    details.forEach(detail => {
+      const tr = createAndAppend('tr', tBody, { class: `${detail.toLowerCase()}-row` });
+      createAndAppend('td', tr, {
+        text: `${detail}: `,
+        class: `label ${detail.toLowerCase()}-head`,
+      });
+      createAndAppend('td', tr, { class: `${detail.toLowerCase()}-data` });
+    });
+    const repoName = document.querySelector('.repository-data');
+    createAndAppend('a', repoName, {
+      href: hyfRepo.html_url,
+      target: '_blank',
+      text: hyfRepo.name,
+      class: 'repo-name',
+    });
+    if (hyfRepo.description === null) {
+      document.querySelector('.description-row').setAttribute('class', 'hide-description-row');
+    } else {
+      document.querySelector('.description-data').innerText = hyfRepo.description;
+    }
+    document.querySelector('.forks-data').innerText = hyfRepo.forks;
+    document.querySelector('.updated-data').innerText = new Date(
+      hyfRepo.updated_at,
+    ).toLocaleString();
+  }
```

Comment:
This function deals adequately with its task. However, you are creating CSS classes in the DOM just for the purpose of getting access to the elements concerned later in your code. You don't need these classes for styling. While this works, I would prefer to stay completely in the JavaScript domain, for instance using code like this:

```js
function createHyfRepoDetails(leftDiv, hyfRepo) {
  const table = createAndAppend('table', leftDiv, { class: 'table' });
  const tBody = createAndAppend('tbody', table);
  const tableRows = {
    name: { label: 'Repository' },
    description: { label: 'Description' },
    forks: { label: 'Forks' },
    updated: { label: 'Updated' },
  };
  Object.keys(tableRows).forEach(key => {
    const row = tableRows[key];
    const tr = createAndAppend('tr', tBody);
    createAndAppend('td', tr, {
      text: `${row.label}: `,
      class: 'label',
    });
    row.elem = createAndAppend('td', tr);
  });
  createAndAppend('a', tableRows.name.elem, {
    href: hyfRepo.html_url,
    target: '_blank',
    text: hyfRepo.name,
    class: 'repo-name',
  });
  if (hyfRepo.description === null) {
    tableRows.description.elem.setAttribute('class', 'hide-description-row');
  } else {
    tableRows.description.elem.innerText = hyfRepo.description;
  }
  tableRows.forks.elem.innerText = hyfRepo.forks;
  tableRows.updated.elem.innerText = new Date(hyfRepo.updated_at).toLocaleString();
}
```

Reply:
okay , thanks